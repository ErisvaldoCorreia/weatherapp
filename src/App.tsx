import React from 'react';
import { View } from 'react-native';
import { HomeScreen } from './pages';

export default function App() {
  return (
    <View testID='app'>
      <HomeScreen />
    </View>
  );
}

