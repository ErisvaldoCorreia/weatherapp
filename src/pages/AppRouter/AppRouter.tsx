import React from 'react';
import { createStackNavigator } from '@react-navigation/stack';
import { NavigationContainer } from '@react-navigation/native';

import { HomeScreen } from '../';
import { WeatherScreen } from '../';

export type RootStackParamList = {
  Home: undefined;
  Weather: { latitude: number; longitude: number };
};

const RootStack = createStackNavigator<RootStackParamList>();

const AppRouter = () => {
  return (
    <NavigationContainer>
      <RootStack.Navigator screenOptions={{headerShown: false}}>
        <RootStack.Screen name="Home" component={HomeScreen} />
        <RootStack.Screen name="Weather" component={WeatherScreen} />
      </RootStack.Navigator>
    </NavigationContainer>
  );
}

export default AppRouter;