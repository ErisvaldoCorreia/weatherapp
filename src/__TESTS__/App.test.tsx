import React from 'react';
import { View } from 'react-native';
import { render } from '@testing-library/react-native';

import AppRouter from '../pages/AppRouter';
import App from '../App';

jest.mock('../pages/AppRouter', () => jest.fn());

describe('Testing App Page', () => {
    it('Should render routes', () => {
        (AppRouter as jest.Mock).mockReturnValueOnce(
            <View testID='mock-routes' />
        );

        const wrapper = render(<App />);
        wrapper.getByTestId('mock-routes');
    });
});
