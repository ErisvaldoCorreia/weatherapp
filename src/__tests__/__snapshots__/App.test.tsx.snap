// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Testing App Should match with snapshot of the screen App 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#fff",
      "flex": 1,
      "justifyContent": "center",
    }
  }
  testID="app"
>
  <Text>
    Open up App.tsx to start working on your app!
  </Text>
</View>
`;
